### PR TITLE
Enhance/optimize ip comparison

### PR DESCRIPTION
**Pull request by jsakamoto**

### CODE REVIEW

File: IPAddressRange/Bits.cs [hunk 1]
```diff
@@ -1,40 +1,72 @@
...
+        public static bool LtE(byte[] A, byte[] B, int offset = 0)
         {
-            return A.Zip(B, (a, b) => a == b ? 0 : a < b ? 1 : -1)
-                .SkipWhile(c => c == 0)
-                .FirstOrDefault() >= 0;
+            for (var i = 0; i < A.Length; i++)
+            {
+                if (A[i] != B[i]) return A[i] <= B[i];
+            }
+            return true;
```
Comment (jsakamoto) [inline IPAddressRange/Bits.cs]: @Menur Hi, 

I found that the 3rd argument `int offset` is not used anywhere.

I think, `offset` should be used like this:

```
for (var i = offset; ...)
```

and also, adding argument check is more better, I think.

- `offset` is less than 0, or not.
- `offset` is greater than `A.length`, or not.

Comment (jsakamoto) [inline IPAddressRange/Bits.cs]: @Menur 

Ah, one more thing.

This method expect that A and B is same length.

I think, If the length of A is larger than of B, this method will crash.

Comment (Menur) [reply IPAddressRange/Bits.cs]: Oh, I forgot to add that. i should be definitely set to offset.

Comment (Menur) [reply IPAddressRange/Bits.cs]: To persist same behavior as LINQ `Math.Min(A.Length, B.Length)` should be used for the `max i`.

**Pull request merged**